930-delegate-tasks/src/main/java/io/harness/delegate/k8s/K8sTrafficRoutingRequestHandler.java

```
@@ -209,15 +232,75 @@ void applyTrafficRouting(K8sDelegateTaskParams k8sDelegateTaskParams, ILogStream
-    if (trafficRoutingConfig.isNormalizationNeeded()) {
-      List<Triple<String, Integer, Integer>> destinationsNormalizationInfo =
-          trafficRoutingConfig.getDestinationsNormalizationInfo();
-      logCallback.saveExecutionLog("Configured destinations weights will be normalized:", WARN);
-      destinationsNormalizationInfo.stream().forEach(triplet
-          -> logCallback.saveExecutionLog(format("[%s] destination weight [%s] will be normalized to value [%s].",
-              color(triplet.getLeft(), Yellow, Bold), color(String.valueOf(triplet.getMiddle()), Yellow, Bold),
-              color(String.valueOf(triplet.getRight()), Yellow, Bold))));
+  Optional<String> prepareTrafficRoutingPatch(K8sInfraDelegateConfig k8sInfraDelegateConfig,
+      K8sTrafficRoutingConfig trafficRoutingConfig, TrafficRoutingInfoDTO trafficRoutingInfo,
+      ILogStreamingTaskClient logStreamingTaskClient, CommandUnitsProgress commandUnitsProgress)
+      throws JsonProcessingException {
+    LogCallback logCallback =
+        k8sTaskHelperBase.getLogCallback(logStreamingTaskClient, TrafficRouting, true, commandUnitsProgress);
+
+    logCallback.saveExecutionLog(
+        format("Preparing to update Traffic Routing resource %s based on inherited configuration.%n",
+            trafficRoutingInfo.getName()));
+
+    Object trafficRoutingClusterResource = kubernetesContainerService.getCustomObject(kubernetesConfig,
+        trafficRoutingInfo.getName(), k8sInfraDelegateConfig.getNamespace(), trafficRoutingInfo.getPlural(),
+        K8sApiVersion.fromApiVersion(trafficRoutingInfo.getVersion()));
+
+    TrafficRoutingResourceCreator trafficRoutingResourceCreator =
+        k8sTrafficRoutingCreators.get(trafficRoutingInfo.getPlural());
+
+    Optional<String> patch = trafficRoutingResourceCreator.generateTrafficRoutingPatch(
+        trafficRoutingConfig, trafficRoutingClusterResource, logCallback);
+
+    if (patch.isEmpty()) {
```

I would suggest to throw the exception instead of returning empty, will be impossible for customer to figure out the reason behind failure without the context of the original exception